packages/icons/src/icon/index.js

```diff
@@ -3,6 +3,20 @@
  */
 import { cloneElement } from '@wordpress/element';
 
+// Disable reason: JSDoc linter doesn't seem to parse the union (`&`) correctly.
+/* eslint-disable jsdoc/valid-types */
+/** @typedef {{icon: string} & {size?: number} & import('@wordpress/primitives').ComponentPropsWithoutRef<'SVG'>} IconProps */
```

Comment:
`icon` type should be `JSX.Element` or something along those lines.

```suggestion
/** @typedef {{icon: JSX.Element} & {size?: number} & import('@wordpress/primitives').ComponentPropsWithoutRef<'SVG'>} IconProps */
```

Reply:
Of course, thanks :facepalm: 

Think it's fine to start even more specific? Using SVG element as the type?

Reply:
> Think it's fine to start even more specific? Using SVG element as the type?

I'd say it's fine as it is. We might be able to get to a more narrow type, but it would require some additional changes and may not be worth the effort. All the icons in `library` are typed `JSX.Element`. It may be possible to change `SVG` in primitives to return something unique, but it may be more effort than it's worth at this time.